### PR TITLE
feat(data_structures): add Transaction::new factory method

### PR DESCRIPTION
**Pull request by anler**
The Transaction::new factory method takes inputs, outputs and a signing key and returns a Transaction object containing those inputs and outputs and with the appropriate signatures. Since keys and signatures needs to be implemented first in witnet_crypto crate, the method returns dummy signatures.

### CODE REVIEW

File: data_structures/src/chain.rs [hunk 1]
```diff
@@ -8,6 +8,9 @@ use std::convert::AsRef;
 use std::fmt;
 use witnet_crypto::hash::{calculate_sha256, Sha256};
 
+/// Current version used when creating [`Transaction`](Transaction)s
+const TRANSACTION_VERSION: u32 = 0;
```
Comment (aesedepece) [inline data_structures/src/chain.rs]: I would add here:
```rust
// TODO: move TRANSACTION_VERSION into a consensus constant
```

File: data_structures/src/chain.rs [hunk 2]
```diff
@@ -238,6 +241,32 @@ pub struct Transaction {
...
+        for _ in inputs.iter() {
+            signatures.push(KeyedSignature {
+                signature: Signature::Secp256k1(Secp256k1Signature {
+                    r: [0; 32],
+                    s: [0; 32],
+                    v: 0,
+                }),
+                public_key: [0; 32],
+            })
+        }
```
Comment (aesedepece) [inline data_structures/src/chain.rs]: Why not functional? :stuck_out_tongue_closed_eyes: 

```rust
let signatures = inputs.iter()
    .map(|_input| KeyedSignature {
        signature: Signature::Secp256k1(Secp256k1Signature {
            r: [0; 32],
            s: [0; 32],
            v: 0,
        }),
        public_key: [0; 32],
    })
    .collect();
```

File: data_structures/src/chain.rs (hunk 1 above)
Comment (anler) [reply data_structures/src/chain.rs]: Then I'm removing the constant and passing it as a parameter to the factory

File: data_structures/src/chain.rs (hunk 2 above)
Comment (anler) [reply data_structures/src/chain.rs]: I was worried that `.collect()` didn't create a vector of the right capacity and thus were a bit less performant, but I was wrong, [this is how it is implemented](https://doc.rust-lang.org/src/alloc/vec.rs.html#1782)

**Pull request merged**